### ISSUE TITLE
Running with non-root user

### ISSUE THREAD

**Issue by jsimdb** [selkies-project/docker-nvidia-glx-desktop#36]
I'm trying to create a chrome container based on this image (for rendering browser output headlessly), but am running into permission problems. Here's an example Dockerfile:

```
FROM nvidia/cudagl:11.2.2-runtime-ubuntu18.04

RUN groupadd -g 1000 testgroup && \
    useradd -ms /bin/bash testuser -u 1000 -g 1000 && \
    usermod -a -G adm,audio,cdrom,dialout,dip,fax,floppy,lp,plugdev,sudo,tape,tty,video,voice testuser

USER testuser
ENTRYPOINT "/bin/bash"
```

Running `nvidia-smi` gives the following error: `Failed to initialize NVML: Insufficient Permissions`

My application uses VirtualGL and Xvfb to render Chrome with a GPU if that's relevant. Works perfectly fine with the root user.

What am I missing here? From what I've read, the group `video` should be adequate, but this is not the case.

Thanks for the great repo! It's been a big help.

**Comment by jsimdb** [comment]
TL;DR - Check the gid of `vglusers` group on the host. Add this group with the gid in the container, and add the user to this group.

So investigating this a bit, I looked at the nvidia devices in the container:

```
root@56cef279b83f:/# cd /dev
root@56cef279b83f:/dev# ls -l | grep nvidia
crw-rw---- 1 root 1005 195,   0 Nov 23 23:13 nvidia0
crw-rw---- 1 root 1005 195, 255 Nov 23 23:13 nvidiactl
crw-rw---- 1 root 1005 195, 254 Nov 23 23:13 nvidia-modeset
crw-rw-rw- 1 root root 506,   0 Nov 23 23:13 nvidia-uvm
crw-rw-rw- 1 root root 506,   1 Nov 23 23:13 nvidia-uvm-tools
```

The nvidia devices belonged to a group with gid 1005. This was odd as there was no group in the container with that ID. 

I went to look into the devices on the host, and as per my VGL setup, they belong to `root`, or the `vglusers` group.

```
(venv) jsim@goliath:/var/log$ cd /dev/
(venv) jsim@goliath:/dev$ ls -l | grep nvidia
crw-rw----   1 root vglusers 195,   0 Nov 24 10:13 nvidia0
drwxr-xr-x   2 root root           80 Nov 24 10:31 nvidia-caps
crw-rw----   1 root vglusers 195, 255 Nov 24 10:13 nvidiactl
crw-rw----   1 root vglusers 195, 254 Nov 24 10:13 nvidia-modeset
crw-rw-rw-   1 root root     506,   0 Nov 24 10:13 nvidia-uvm
crw-rw-rw-   1 root root     506,   1 Nov 24 10:13 nvidia-uvm-tools
```

As it turns out, `vglusers` has a gid of 1005! 

```jsim@goliath:/dev$ cat /etc/group | grep vglusers
vglusers:x:1005:jsim
```

So in my Dockerfile, all I had to do is add the group `vglusers` with gid 1005, and add my user to this group. Problem solved.

```
RUN groupadd -g 1005 vglusers && \
    useradd -ms /bin/bash testuser -u 1000 -g 1005 && \
    usermod -a -G video,sudo testuser
```

**Comment by ehfd** [comment]
Great thanks for the feedback. This will be in my backlogs.

**Comment by ehfd** [comment]
Also note that Chromium browsers will be properly supported in VirtualGL 3.1. https://github.com/VirtualGL/virtualgl/releases/tag/3.1beta1 We are not there yet.

**Comment by jsimdb** [comment]
This is my own image which is a stripped down version of your glx repo (only installing Vulkan & virtualGL). I think the problem is more to do with the host configuration than the container. Useful information for troubleshooting if others run into the same issue.

**Comment by ehfd** [comment]
Try the 3.1 beta, then use EGL translation instead of bringing Xorg to the host. It should be cleaner.

**Comment by ehfd** [comment]
Check this as well, https://docs.cerit.io/docs/limitations.html#consequences written by @xhejtman.

**Comment by xhejtman** [comment]
What we do now is the following:
```
FROM ubuntu:jammy as ub

ENV LANG en_US.UTF-8
ENV LANGUAGE en_US:en
ENV LC_ALL en_US.UTF-8

RUN apt-get update && apt-get -y dist-upgrade &&  apt-get -y install fakeroot tzdata locales && apt-get clean && rm -rf /var/lib/apt/lists/*

RUN ln -fs /usr/share/zoneinfo/Europe/Prague /etc/localtime && dpkg-reconfigure --frontend noninteractive tzdata && sed -i -e 's/# en_US.UTF-8 UTF-8/en_US.UTF-8 UTF-8/' /etc/locale.gen && locale-gen && rm -f /usr/bin/sudo; ln -s /usr/bin/fakeroot /usr/bin/sudo

FROM scratch

COPY --chown=1000:1000 --from=ub / /

USER 1000

ENV LANG en_US.UTF-8
ENV LANGUAGE en_US:en
ENV LC_ALL en_US.UTF-8
```

**Comment by ehfd** [comment]
A few other things to note myself:
- Check NVIDIA drivers
- Use PipeWire over PulseAudio server
- Use KasmVNC
- Try VirtualGL 3.1.1 beta for egl
- Docker-Compose
- Check fakeroot usage
